sdk/src/types/block/output/delegation.rs

```diff
@@ -359,7 +359,7 @@ impl DelegationOutput {
 
     // Transition, just without full ValidationContext.
     pub(crate) fn transition_inner(current_state: &Self, next_state: &Self) -> Result<(), StateTransitionError> {
-        if !(current_state.delegation_id.is_null() && !next_state.delegation_id().is_null()) {
+        if !current_state.delegation_id.is_null() || next_state.delegation_id().is_null() {
```

Comment:
I know it's clippy but tbh I'd allow it, I'd prefer it to read exactly as the TIP is

Reply:
Alright, that's fair